### PR TITLE
GEOMESA-1912 Adding progress indicator for reduce phase of fs ingest

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

~note: stacks on top of #1605~

### CODE REVIEW

File: docs/user/filesystem/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+FileSystem Datastore Architecture
+=================================
+
+The GeoMesa FileSystem Datastore (GeoMesa FSDS) takes advantage of the performance characteristics of modern
+cloud-native and distributed filesystems to scale bulk analytic queries. The FSDS is a good choice for doing bulk egress
+queries or large analytic jobs using frameworks such as Spark SQL and MapReduce. The FSDS differs from other queries
+in that ingest latency is traded for high-throughput query. The FSDS pairs well with low-latency ingest systems such as
+HBase or Kafka via the GeoMesa Lambda datastore to provide an optimal pairing of "hot", "warm" storage options.
```
Comment (aheyne) [inline docs/user/filesystem/architecture.rst]: s/,/ and/

File: docs/user/filesystem/architecture.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+* **HDFS** - Hadoop Distributed File System
+* **S3** - Amazon Simple Storage
+* **GCS** - Google Cloud Storage
+* **WASB** - Windows Azure Blob Store
+
+Choosing a filesystem depends generally on cost and performance requirements. One thing to note is that S3, GCS, and
+WASB are all "cloud-native" storage meaning that they are built into Amazon, Google, and Microsoft Azure cloud
+platforms. These cloud-native filesystems are scaled separately from the compute nodes which generally provides a more
+cost efficient storage solution. In general their price per GigaByte of storage is lower then using HDFS though they
+may be slightly more latent. They also have the ability to persist data when after you turn off all your compute nodes.
```
Comment (aheyne) [inline docs/user/filesystem/architecture.rst]: `when after` one or the other.

File: docs/user/filesystem/architecture.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Any of the filesystems mentioned about are good choices for the FSDS. If you have more questions about making a choice
```
Comment (aheyne) [inline docs/user/filesystem/architecture.rst]: s/about/above/

File: docs/user/filesystem/architecture.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+contact the GeoMesa team.
+
+Partition Schemes
+-----------------
+
+The partition scheme defines how data is stored within the filesystem. The scheme is important because it defines how
+the data is queried. Most OLAP queries in GeoMesa contain a date range and geometric predicate and the partition scheme
+can aide in finding the files satisfy the query. There are two main partition schmes used with GeoMesa FSDS:
```
Comment (aheyne) [inline docs/user/filesystem/architecture.rst]: s/satisfy/to satisfy/

File: docs/user/filesystem/architecture.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+may be slightly more latent. They also have the ability to persist data when after you turn off all your compute nodes.
+
+Any of the filesystems mentioned about are good choices for the FSDS. If you have more questions about making a choice
+contact the GeoMesa team.
+
+Partition Schemes
+-----------------
+
+The partition scheme defines how data is stored within the filesystem. The scheme is important because it defines how
+the data is queried. Most OLAP queries in GeoMesa contain a date range and geometric predicate and the partition scheme
```
Comment (aheyne) [inline docs/user/filesystem/architecture.rst]: [opinion] I break this sentence up.
`Most OLAP queries in GeoMesa contain a date range and geometric predicate. The partition scheme...`

File: docs/user/filesystem/commandline.rst [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+--------------------
+
+To build from source make sure you have Maven, Git, and Java SDK 1.8 installed. To build just the GeoMesa FileSystem
+distribution do this:
+
+.. code-block:: bash
+
+    git clone git@github.com:locationtech/geomesa.git
+    cd geomesa
+    build/mvn clean install -pl geomesa-fs/geomesa-dist -am -T4
```
Comment (aheyne) [inline docs/user/filesystem/commandline.rst]: `-pl geomesa-fs/geomesa-fs-dist` or `-pl :geomesa-fs-dist_2.11`

File: docs/user/filesystem/example.rst [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+    $ bin/geomesa-fs  export -p file:///tmp/dstest -e parquet -f example-csv
+
+    id,fid:Integer:index=false,name:String:index=true,age:Integer:index=false,lastseen:Date:default=true:index=false,*geom:Point:srid=4326
+    26236,26236,Hermione,25,2015-06-07T00:00:00.000Z,POINT (40.232 -53.2356)
+    3233,3233,Severus,30,2015-10-23T00:00:00.000Z,POINT (3 -62.23)
+    23623,23623,Harry,20,2015-05-06T00:00:00.000Z,POINT (-100.236523 23)
+    INFO  Feature export complete to standard out in 1676ms for 3 features
+
+Now lets inpsect the filesystem to see what it looks like:
```
Comment (aheyne) [inline docs/user/filesystem/example.rst]: s/inpsect/inspect/

File: docs/user/filesystem/geoserver.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        * hadoop-mapreduce-client-core-2.7.3.jar
+        * hadoop-yarn-api-2.7.3.jar
+        * hadoop-yarn-common-2.7.3.jar
+        * commons-configuration-1.6.jar
+
+        You can use the bundled ``$GEOMESA_FS_HOME/bin/install-hadoop.sh`` script to install these JARs.
+
+The FileSystem data store requires the configuration file ``core-site.xml`` to be on the classpath. This can
+be accomplished by placing the file in ``geoserver/WEB-INF/classes`` (you should make the directory if it
+doesn't exist). Utilizing a symbolic link will be use full here so any changes are reflected in GeoServer.
```
Comment (aheyne) [inline docs/user/filesystem/geoserver.rst]: s/use full/useful/

File: docs/user/filesystem/index.rst [hunk 9]
```diff
@@ -0,0 +1,22 @@
+FileSystem Datastore
+====================
+
+The GeoMesa FileSystem Datastore provides a cost-effective, performant solution for massive OLAP-style spatial
+data analytics using frameworks such as Apache Spark. It utilizes modern columnar formats that enable data compression
+and column-based encodings for efficient OLAP queries. GeoMesa DFDS can run on top of any distributed or local
```
Comment (aheyne) [inline docs/user/filesystem/index.rst]: s/DFDS/FSDS/

File: docs/user/filesystem/index.rst [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+filesystem including Amazon S3, Hadoop HDFS, Google FileStorage, and Azure BlobStore. This enables users to realize
+cost savings through the utilization of elastic compute resources in lieu of dedicated servers
```
Comment (aheyne) [inline docs/user/filesystem/index.rst]: Period

File: docs/user/filesystem/index.rst [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+The FileSystem datastore can be used to ETL new data into a filesystem or  wrap existing file system storage directories
```
Comment (aheyne) [inline docs/user/filesystem/index.rst]: [picky] extra space

**Comment by elahrvivaz** [comment]
rebased

**Comment by jahhulbert-ccri** [comment]
Retested with local FSDS and it worked

**Pull request merged**